RockLib.Encryption.Tests/SymmetricCryptoTests.cs

```diff
@@ -69,17 +85,27 @@ public void CanCreateSymmetricCryptoByEncryptionSettings()
         [Test]
         public void CanEncryptDecryptByRepoByEncryptionSettings()
         {
-            var credentialMock = new Mock<ICredential>();
-            credentialMock.Setup(cm => cm.Algorithm).Returns(SymmetricAlgorithm.Aes);
-            credentialMock.Setup(cm => cm.IVSize).Returns(16);
-            credentialMock.Setup(cm => cm.GetKey()).Returns(GetSequentialByteArray(16));
+            var builder = new ConfigurationBuilder();
+            builder.AddInMemoryCollection(new Dictionary<string, string>
+            {
+                {"Section:Value", "1J9Og/OaZKWdfdwM6jWMpvlr3q3o7r20xxFDN7TEj6s="}
+            });
 
             var crypto = new SymmetricCrypto();
             crypto.EncryptionSettings = new CryptoConfiguration
             {
-                Credentials = new List<ICredential>
+                Credentials = new List<Credential> 
                 {
-                    credentialMock.Object
+                    new Credential
```

Comment:
The part here with LateBoundConfigurationSection is so awkward. I wonder if we should add a constructor to the Credential class that hides that complexity. I think your implementation here would be the right implementation for the constructor too.

``` c#
public Credential(SymmetricAlgorithm algorithm, int ivSize, byte[] key)
{
    // TODO: argument checking

    Algorithm = algorithm;
    IVSize = ivSize;

    var builder = new ConfigurationBuilder();
    builder.AddInMemoryCollection(new Dictionary<string, string> { { "Section:Value", Convert.ToBase64String(key) } });

    Key = new LateBoundConfigurationSection<IProtectedValue>
    {
        Type = "RockLib.DataProtection.UnprotectedBase64Value, RockLib.DataProtection",
        Value = builder.Build().GetSection("Section")
    };
}
```

Thechange would mean this line would simply be this:

``` c#
new Credential(SymmetricAlgorithm.Aes, 16, Convert.FromBase64String("1J9Og/OaZKWdfdwM6jWMpvlr3q3o7r20xxFDN7TEj6s=")
```